Update Tailwind Docs to Tailwind v4 (#1065)
solidjs/solid-docs: src/routes/guides/styling-components/tailwind.mdx

mainNavExclude: true
---

> [Tailwind CSS](https://tailwindcss.com/) is an on-demand utility CSS library that integrates seamlessly with Solid as a built-in PostCSS plugin. For **Tailwind CSS v3** refer to [TailwindCSS v3](/guides/styling-components/tailwind-v3).

Reviewer: ```suggestion
<Callout type="info" title="Note">

This guide is for TailwindCSS v4. For **Tailwind CSS v3** refer to [TailwindCSS v3](/guides/styling-components/tailwind-v3).

</Callout>

[Tailwind CSS](https://tailwindcss.com/) is an on-demand utility CSS library that integrates seamlessly with Solid as a built-in PostCSS plugin. 
```

I think it's better to have this as a callout at the very beginning.

In the current version, people who want v3 docs have to read the introduction line twice.

## Installation

> 1. Install Tailwind CSS as a dependency:

Reviewer: ```suggestion
1. Install Tailwind CSS as a development dependency:
```

I'm not sure why tailwind docs suggests to install it as a production dependency. But I believe installing it as a dev dependency is better. It doesn't have any use-case in production AFAIK. The related dependncies (vite and postcss) are also installed as a dev dependency.

<TabsCodeBlocks>

> <div id="npm">

Reviewer: ```suggestion
<div id="npm">

```

```bash frame="none"

> npm i tailwindcss @tailwindcss/postcss postcss

Reviewer: ```suggestion
npm i --save-dev tailwindcss @tailwindcss/postcss postcss
```

Same as above.

> ```

Reviewer: ````suggestion
```

````

</div>

> 

Reviewer: ```suggestion
```

> <div id="yarn">

Reviewer: ```suggestion
<div id="yarn">

```

```bash frame="none"

> yarn add tailwindcss @tailwindcss/postcss postcss

Reviewer: ```suggestion
yarn add --dev tailwindcss @tailwindcss/postcss postcss
```

Same as above.

> ```

Reviewer: ````suggestion
```

````

</div>

> 

Reviewer: ```suggestion
```

> <div id="pnpm">

Reviewer: ```suggestion
<div id="pnpm">

```

```bash frame="none"

> pnpm add tailwindcss @tailwindcss/postcss postcss

Reviewer: ```suggestion
pnpm add --save-dev tailwindcss @tailwindcss/postcss postcss
```

Same as above.

> ```

Reviewer: ````suggestion
```

````

</div>

> 

Reviewer: ```suggestion
```

> <div id="bun">

Reviewer: ```suggestion
<div id="bun">

```

```bash frame="none"

> bun add tailwindcss @tailwindcss/postcss postcss

Reviewer: ```suggestion
bun add --save-dev tailwindcss @tailwindcss/postcss postcss
```

Same as above.

> ```

Reviewer: ````suggestion
```

````

</div>
</TabsCodeBlocks>

> 2. Add `@tailwind/postcss` to the `plugins`in your PostCSS configuration. If you do not have a PostCSS configuration file, create a new one called `postcss.config.mjs`.

Reviewer: ```suggestion
2. Add `@tailwind/postcss` to the `plugins` in your PostCSS configuration. If you do not have a PostCSS configuration file, create a new one called `postcss.config.mjs`.
```

> ```js

Reviewer: ```suggestion
```js title="postcss.config.mjs"
```

> /* postcss.config.mjs */

Reviewer: ```suggestion
```

export default {
	plugins: {
		"@tailwindcss/postcss": {},
	}
}
```

As of Tailwind CSS v4, **tailwind.config.js** is no longer required.

> 

Reviewer: ```suggestion
```

This is not a migration guide so there is no need to mention that.

For a deeper dive into configuration, you can check out the [Tailwind Official Documentation](https://tailwindcss.com/docs/configuration).

> ## Add Tailwind directives
> 
> In your `src/index.css` file, add the following Tailwind directives:

Reviewer: ```suggestion
## Import TailwindCSS

Add an `@import` to your `src/index.css` file that imports Tailwind CSS.
```

Tailwind v4 uses native CSS imports. It's not tailwind-specific thing.

> ```css

Reviewer: ```suggestion
```css title="src/index.css"
```

@import "tailwindcss";
```

> These directives ensure that Tailwind CSS is properly initialized and establish the correct order for utilities. You can append custom CSS below these directives.

Reviewer: ```suggestion
```

> ## Import TailwindCSS

Reviewer: ```suggestion
## Import your CSS file
```

We probably need to remove this section. For now, because GitHub is bad, I just suggest to change the title since it's the same as the preivous section.

mainNavExclude: true
---

[Tailwind CSS](https://tailwindcss.com/) is an on-demand utility CSS library that integrates seamlessly with Solid as a built-in Vite plugin. For **Tailwind CSS v3** refer to [TailwindCSS v3](/guides/styling-components/tailwind-v3).

## Installation

1. Install Tailwind CSS as a dependency:

<TabsCodeBlocks>
<div id="npm">
```bash frame="none"
npm i tailwindcss @tailwindcss/vite
```
</div>

<div id="yarn">
```bash frame="none"
yarn add tailwindcss @tailwindcss/vite
```
</div>

<div id="pnpm">
```bash frame="none"
pnpm add tailwindcss @tailwindcss/vite
```
</div>

<div id="bun">
```bash frame="none"
bun add tailwindcss @tailwindcss/vite
```
</div>
</TabsCodeBlocks>

> 2. Add the @tailwindcss/vite plugin to your SolidJS Vite configuration.

Reviewer 2: I'd recommend staying with the postcss config for tailwind 4. It's a lot less flaky with solid-start

Author: > I'd recommend staying with the postcss config for tailwind 4. It's a lot less flaky with solid-start

I have updated it now to use PostCSS plugin instead of Vite plugin
